Mathlib/Topology/Compactness/Lindelof.lean

```diff
@@ -177,6 +176,29 @@ theorem IsLindelof.elim_nhds_subcover (hs : IsLindelof s) (U : X → Set X)
   · have : ⋃ x ∈ t, U ↑x = ⋃ x ∈ Subtype.val '' t, U x := biUnion_image.symm
     rwa [← this]
 
+/-- For every nonempty open cover of a Lindelöf set, there exists a subcover indexed by ℕ. -/
+theorem IsLindelof.indexed_countable_subcover {ι : Type v} [Nonempty ι]
+    (hs : IsLindelof s) (U : ι → Set X) (hUo : ∀ i, IsOpen (U i)) (hsU : s ⊆ ⋃ i, U i) :
+    ∃ f : ℕ → ι, s ⊆ ⋃ n, U (f n) := by
+  rcases IsLindelof.elim_countable_subcover hs U hUo hsU with ⟨c, ⟨c_count, c_cov⟩⟩
```

Comment:
You can also write
```lean
obtain ⟨c, ⟨c_count, c_cov⟩⟩ := IsLindelof.elim_countable_subcover hs U hUo hsU
```
There is no clear preference, but the `obtain` keyword makes it clear that you are constructing something. At any rate, obtain is an `rcases` after all, so it's up to you.

Reply:
Yes - @grunweg pointed this tactic out where `have` was used similarly in some code I (didn't write and) moved around above. I much prefer the word `obtain` for making things clear here. Thanks!